torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -155,10 +92,107 @@ def _pre_load_state_dict_hook(
         _replace_by_prefix(state_dict, prefix, prefix + f"{_CHECKPOINT_PREFIX}.")
 
 
+class OffloadWrapper(ActivationWrapper):
+    def __init__(self, mod):
+        super().__init__(mod)
+
+    def forward(self, *args, **kwargs):
+        with save_on_cpu(pin_memory=True):
+            return self._checkpoint_wrapped_module(*args, **kwargs)
+
+
+class CheckpointWrapper(ActivationWrapper):
+    """
+    An nn.Module that wraps another nn.Module with checkpointing. Note that this
```

Comment:
nit: Do we want these in code type?
```suggestion
    An ``nn.Module`` that wraps another ``nn.Module`` with checkpointing. Note that this
```